### DISCUSSION TITLE
How to disable reCAPTCHA verification for firebase authentication on iOS?

### DISCUSSION THREAD

**Discussion by sarohabhawna** [firebase/flutterfire#3381] [Q&A]
I am working on a flutter app which has a registration flow like this.

- The user selects their Google/Apple id depending on the device
- Then they enter their phone number to get an SMS
- Then they enter the SMS code to verify it
- Then they enter their personal info and enter into the app

This is the flow on Android but on iOS after entering the phone number the in-app browser open a link for a reCAPTCHA verification. I want to disable it as it results in bad user experience.

So, after looking online I enabled push notification and background mode but I am still getting redirected.

There might be something that I missed but can't find what is it.

Any help would be great. Thanks for your time!

**Comment by Ehesp** [comment]
Take a look here: https://firebase.google.com/docs/auth/ios/phone-auth

You need to ensure that the `Start receiving silent notifications` setup is complete for your app. If it's not, you'll be prompted for the reCAPTCHA flow. 

> So, after looking online I enabled push notification and background mode but I am still getting redirected.

This means you haven't set it up correctly.

**Reply by sarohabhawna** [reply]
Don't know what is going wrong as these steps are already done. In that case, it should work.

![Screen Shot 2020-08-28 at 5 15 57 PM](https://user-images.githubusercontent.com/45165486/91557610-689e0d00-e952-11ea-8ae6-0f3f4ef0bc64.png)

**Reply by Ehesp** [reply]
Try checking logs - there isn't anything special FlutterFire does for this side of things, it's all handled via the iOS SDK.

**Reply by Salakar** [reply]
This guide should help ensuring FCM & APNs is setup correctly: https://firebase.flutter.dev/docs/messaging/ios-integration

**Reply by Dantusaikamal** [reply]
Any way we can do this without running Xcode? I'm using Windows and I could off reCaptcha for Android but couldn't get it working for ios devices

**Reply by satyamraj123** [reply]
anyone got this issue fixed? im having the same issue

**Reply by nilsreichardt** [reply]
I have set up FCM & APNs. Firebase Cloud Messing is working great. So this shows us, that APNs are correct configured. But Firebase Phone Auth still opens every time on iOS the reCAPTCHA page :/

**Reply by anasinfo1** [reply]
the same problem guys !

**Comment by satyamraj123** [comment]
@sarohabhawna  got this issue fixed? im having the same issue

**Reply by dhananjay0610** [reply]
@satyamraj123   got and?

**Comment by dhananjay0610** [comment]
Got the answer?.. I am also having same problem

**Reply by dhananjay0610** [reply]
@sarohabhawna

**Comment by moksh-mahajan** [comment]
1. Go to Apple Development Centre.
2. Under Certficate, Identifier & Profiles, select Keys tab.
3. Click on + icon to register a new key. Give it a name and enable Apple Push Notification Service.
4. Download this .p8 file.
5. Go to your firebase project settings and under Cloud Messaging Tab add this APNs .p8 file for your iOS app and that's it.

**Comment by irfanadot** [comment]
Try to set `true` to `appVerificationDisabledForTesting` while using Firebase phone auth
```

 FirebaseAuth auth = FirebaseAuth.instance;
      await auth.setSettings(appVerificationDisabledForTesting: true); // <-- here is the magic 
      await auth.verifyPhoneNumber(
        phoneNumber: phoneNumber,
     );
```

**Reply by harshnagalla** [reply]
This is not a long term solution as after sometime it gives -  Error getting document: [Error: [auth/unknown] MISSING_CLIENT_IDENTIFIER]

**Reply by naderhezzy** [reply]
This will disable Firebase verification, which means no OTP will be sent to anyone. Only testing numbers with dummy OTP will work.

**Comment by baimamboukar** [comment]
Anyone finally found a solution on this ?

**Comment by HassAlli** [comment]
The issue is still there!

**Reply by anasinfo1** [reply]
yes

**Comment by ahmed-tarek-salem** [comment]
I have the same issue. the FCM is working great and I receive all notifications on my iPhone, but when it comes to the top it still redirects me to the verification tab in the browser.

I checked every single line in the docs and applied it with no luck.

**Comment by Zoudafnan** [comment]
Anyone fixed this ?

**Comment by prabakaranr** [comment]
Any fix for this? I am using flutter web app with phone number OTP verification.

**Comment by mhmzdev** [comment]
Anything folks? Still same issue although I've the .p8 key in cloud messaging tab at firebase project

**Comment by BrawlerXull** [comment]
Can anyone help me with #12525

**Comment by FujiKinaga** [comment]
How about checking if `Remote_notifications` is set in Background Modes?

<img width="500" alt="スクリーンショット 2024-03-27 15 12 49" src="https://github.com/firebase/flutterfire/assets/27717556/f6e92db4-d441-4390-b152-811d3acc68aa">

**Reply by uxmarkowski** [reply]
Hi, I finally solve problem for myself. 
Enable AppCheck (AppAttest) in Firebase.
Enable remote notifications, background proccessing in XCODE
Change Firebase Auth to Identify Platform Status
Set APNS key in Firebase
Set AppAttest check in bundle identifier in developer.apple.com
Make this in main.dart 
  await FirebaseAppCheck.instance.activate(
    webProvider: ReCaptchaV3Provider('recaptcha-v3-site-key'),
    androidProvider: AndroidProvider.debug,
    appleProvider: AppleProvider.appAttest,
  );
  
And do this before send OTP
 await _auth.setSettings(appVerificationDisabledForTesting: true);

**Reply by Aarush-Acharya** [reply]
What do you mean when you say "Change Firebase Auth to Identify Platform Status"

**Reply by Aarush-Acharya** [reply]
@uxmarkowski  Also you mentioned background proccessing that demands me of mentioning in your Info.plist  the background task scheduler identifier, so what all identifiers am i to mention

**Comment by BrawlerXull** [comment]
can you guys please look into #12525

**Comment by Sudarshanshiva** [comment]
can anyone tell me how to get out of this recaptcha in jetpack compose

**Comment by taku126189** [comment]
Any update?
